plugins/coverimage.koplugin/main.lua

```diff
@@ -158,16 +197,440 @@ function CoverImage:onReaderReady(doc_settings)
+                    os.remove(file)
+            end
+        end
+    end
+end
+
+function CoverImage:getCacheFiles(cache_path, cache_prefix)
+    local cache_count = 0
+    local cache_size_KiB = 0
+    local files = {}
+    for entry in lfs.dir(self.cover_image_cache_path) do
+        if entry ~= "." and entry ~= ".." then
+            local file = cache_path .. entry
+            if entry:sub(1,self.cover_image_cache_prefix:len()) == cache_prefix
+                and lfs.attributes(file, "mode") == "file" then
+                cache_count = cache_count + 1
+                files[cache_count] = {
+                    name = file,
+                    size = math.floor((lfs.attributes(file).size + 1023) / 1024), -- round up to KiB
+                    mod = lfs.attributes(file).modification,
+                }
+                cache_size_KiB = cache_size_KiB + files[cache_count].size -- size in KiB
+            end
+        end
+    end
+    logger.dbg("CoverImage: start - cache size: ".. cache_size_KiB .. " KiB, cached files: " .. cache_count)
+    return cache_count, cache_size_KiB, files
+end
+
+function CoverImage:cleanCache()
+    if not self:isCacheEnabled() then
+        self:emptyCache()
+        return
+    end
+
+    local cache_count, cache_size_KiB, files = self:getCacheFiles(self.cover_image_cache_path, self.cover_image_cache_prefix)
+
+    -- delete the oldest files first
+    table.sort(files, function(a, b) return a.mod < b.mod end)
+    local index = 1
+    while (cache_count > self.cover_image_cache_maxfiles and self.cover_image_cache_maxfiles ~= 0)
+        or (cache_size_KiB > self.cover_image_cache_maxsize * 1024 and self.cover_image_cache_maxsize ~= 0)
+        and index <= #files do
+        os.remove(files[index].name)
+        cache_count = cache_count - 1
+        cache_size_KiB = cache_size_KiB - files[index].size
+        index = index + 1
+    end
+    logger.dbg("CoverImage: clean - cache size: ".. cache_size_KiB .. " KiB, cached files: " .. cache_count)
+end
+
+function CoverImage:isCacheEnabled(path)
+    if not path then
+        path = self.cover_image_cache_path
+    end
+    if Device.isAndroid() then
+        return self.cover_image_cache_maxfiles >= 0 and self.cover_image_cache_maxsize >= 0
+            and lfs.attributes(path, "mode") == "directory"
+            and path ~= "/sdcard/koreader/cache/"  -- interferes with frontent cache-framework
+            and realpath(path) ~= realpath(android.getExternalStoragePath() .. "/koreader/cache/")  -- interferes with frontent cache-framework
+    else
+        return self.cover_image_cache_maxfiles >= 0 and self.cover_image_cache_maxsize >= 0
+            and lfs.attributes(path, "mode") == "directory"
+            and path ~= "./cache/" -- interferes with frontent cache-framework; quick and dirty check
+            and realpath(path) ~= realpath("./cache/") -- interferes with frontent cache-framework; quick and dirty check
```

Comment:
getDataDir()?

Reply:
(Reasoning: IIRC, PWD is not necessarily KOReader's folder on every platform).

Reply:
Yes, this is just for safety. Normally this case should not happen.
But if someone selects a path, and when entering the filename the path could be replaced by `./cache/`
So I would let it in.